### PR TITLE
Display deprecated warning in admin panel pages as well as in the log file

### PR DESCRIPTION
**Pull request by lunny**
This PR includes #26007 's changes but have a UI to prompt administrator about the deprecated settings as well as the log or console warning. Then users will have enough time to notice the problem and don't have surprise like before.

图片

### CODE REVIEW

**Comment by delvh** [comment]
Hmm… We could also think about which parts of #23203 we can reuse.
I think we can reuse it basically completely.
The main problem of that PR is that I still haven't found out how I have to set `LD_FLAGS` so that the version will be updated automatically.

**Comment by lunny** [comment]
> Hmm… We could also think about which parts of #23203 we can reuse. I think we can reuse it basically completely. The main problem of that PR is that I still haven't found out how I have to set `LD_FLAGS` so that the version will be updated automatically.

This is a quick patch to solve the current problem. I think we still need a self-check page and a whole consideration.

**Comment by silverwind** [comment]
image

1. There should be a margin below second box
2. I wonder why font color is different in the two boxes

File: routers/web/web.go [hunk 1]
```diff
@@ -644,7 +645,18 @@ func registerRoutes(m *web.Route) {
...
+		"EnablePackages", setting.Packages.Enabled,
+	), func(ctx *context.Context) {
+		if len(setting.DeprecatedWarnings) > 0 {
+			content := setting.DeprecatedWarnings[0].String()
+			if len(setting.DeprecatedWarnings) > 1 {
+				content += fmt.Sprintf(" (and %d more)", len(setting.DeprecatedWarnings)-1)
+			}
+			ctx.Flash.Error(content)
+		}
+	})
```
Comment (KN4CK3R) [inline routers/web/web.go]: I would like to see this in a method instead inlined.

**Comment by lunny** [comment]
> image
> 
>     1. There should be a `gap`or `margin` below the second box
> 
>     2. I wonder why font color is different in the two boxes

What's the best practice to add a bottom margin for the topbar? The first one is written with js and the second one is written with flash message. And maybe leave those fine-tune to others because I'm not frontend expertise.

File: routers/web/web.go (hunk 1 above)
Comment (lunny) [reply routers/web/web.go]: done.

**Comment by silverwind** [comment]
> What's the best practice to add a bottom margin for the topbar? The first one is written with js and the second one is written with flash message. And maybe leave those fine-tune to others because I'm not frontend expertise.

I'd do a wrapper div, like `class=alert-messages` and this CSS:

```
.alert-messages {
  display: flex;
  flex-direction: column;
  gap: 12px;
}

**Review by wolfogre — APPROVED** [review]

**Review by Zettat123 — APPROVED** [review]

**Pull request merged**